### ISSUE TITLE
[Enhancement] Button to enable/disable all

### ISSUE THREAD

**Issue by DutchPete** [chetachiezikeuzor/MySnippets-Plugin#7]
Sometimes one needs to disable all snippets e.g. when there seems to be a bug somewhere.
It would a time saver not to have to disable each snippet manually.

If there is a button to disable them all, then pressing the button should re-enable them all, and exactly those that were enabled before. So, if snippets 1,2,3, and 5 were enable and snippet 4 disabled, then re-enabling should reset them as before, i.e. 1,2,3, 5 enabled, leave 4 disabled.

**Comment by chetachiezikeuzor** [comment]
Hmm, interesting feature! Though, would it not be a bit of a pain to have the button toggle all snippets that are currently off on? If I wanted to scope out a bug and I used the button to toggle all snippets off, I wouldn't want the snippets that were currently off to then be toggle on. So perhaps there should be a "switch mode" button that switches the status of all snippets, then toggle "on/off" button that turn them all off or on.

**Comment by chetachiezikeuzor** [comment]
My apologies. After reading your request a few times, I realized I am misunderstanding what you're saying. Please correct me if I'm getting this wrong: There are two buttons. One to disable all snippets and one to re-enable all of the snippets that were enabled before. So the re-enable button acts more like an "undo disable" button.

**Comment by DutchPete** [comment]
> There are two buttons. One to disable all snippets and one to re-enable all of the snippets that were enabled before. So the re-enable button acts more like an "undo disable" button.

Yes, that is correct. In my OP I was thinking along the lines of just 1 button to disable them all, while hitting that same button again would re-enable all of the snippets that were enabled before. This to avoid having too many buttons.

If it is only possible with 2 separate buttons, fine. What I am interested in is the concept, and the way you describe it in your 2nd comment above is exactly that.

My apologies for not expressing myself clearly to begin with.

**Comment by churnish** [comment]
I second this! A button to enable/disable all snippets would be super convenient! Right now, toggling each snippet individually takes extra time, especially when managing many at once. A single switch would make testing and organizing snippets a lot smoother."

**Comment by DutchPete** [comment]
@greetclammy Since my request I have made life a lot simpler for myself. I have put all my snippets into 1 `.css` file, 1 big snippet as it were, and in there I have put commented out headers to give a descriptive name to each block of code.

So now, all I have to do is open the settings, Appearance code, scroll down to the snippets section and toggle that 1 big snippet off/on.

I don't actually use Chetachi's plug-in anymore.
